cai2hcl/common.go

```diff
@@ -3,23 +3,22 @@ package cai2hcl
 import (
 	"github.com/GoogleCloudPlatform/terraform-google-conversion/v2/caiasset"
 
-	tfschema "github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/zclconf/go-cty/cty"
 )
 
-// ConvertFunc converts an Asset type into a map[string]interface{} struct.
-type ConvertFunc func(asset *caiasset.Asset) (string, map[string]interface{}, error)
-
-// Converter for individual resource.
-type Converter struct {
-	TFResourceName string
-	Convert        ConvertFunc
-	Resource       *tfschema.Resource
+// Converter interface for resources.
+type Converter interface {
```

Comment:
Switching to interface, so structs that implement the interface can have a member of *tfschema.Resource. 
Since convert function also convert the map[string]interface{} to cty.Value, it needs the schema to proceed. If it is a 
An alternative would be to set tfschema.Resource as a global variable, which is shared in the same package.

Reply:
I'm not sure I understand - it looks like the final struct is basically the same as Converter was previously. It has a way of getting a TF resource name, a conversion function, and a `*tfschema.Resource`, but it had all of those before. Could you elaborate on the benefits of / motivations for this change?

Reply:
convertFunc() if not a struct function, then it need to access some variable outside of that function to get the schema. Since all converters share the same package, that schema will be visible to all resources. I feel the encapsulation is not idea in that way.

Reply:
Converter calls convertFunc() to get cty.Value, and inside convertFunc()  somehow need to fetch the Converter's member to proceed with something, this process looks a bit awkward for me.

Reply:
SGTM - making this an interface seems in line with Go style and allows the Convert function to return a cty value by letting it easily access the TF schema.